R/coh_data_wrangling.R

```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
     vacc_date_col,
     min.len = 1L
   )
-  checkmate::assert_number(
-    immunization_delay,
-    lower = 0, finite = TRUE
+  stopifnot(
+    "Please provide a non-null integer number greater or equal than 0
+    in `outcome_delay`. Use round(`immunization_delay`,0)" =
+      checkmate::test_integerish(immunization_delay, lower = 0, null.ok = FALSE)
   )
   checkmate::assert_names(
     colnames(data),
     must.include = c(vacc_date_col, immunization_date_col)
   )
+  checkmate::assert_date(
+    data[[immunization_date_col]]
+  )
+  # The function can receive one or more column names in vacc_date_col
+  vacc_date_col_ <- vacc_date_col # hard coded to return right error message
+  for (vacc_date_col in c(vacc_date_col_)) {
+    checkmate::assert_date(
+      data[[vacc_date_col]]
+    )
+  }
```

Comment:
Again, confirm whether missing values are allowed in date columns.

Reply:
Allowed.